woocommerce/woocommerce-android
# Select shipping rate

Close: #12286

### Description
This PR supports selecting a shipping rate and the corresponding signature requested. To achieve this goal, I needed to:
1. Update the shipping rates models to include the missing information
2. Refactor the main flow's view model to observe the selected shipping rate and the signature required options
3. Refactor the UI to send shipping rate selection and signature required selections
4. Update the unit tests because of the view model refactor

⚠️ This PR focuses only on the happy path. We will work on error handling on M2  

### Testing information
1. Open the orders section
2. Select an order that is eligible to create shipping labels (processing state)
3. Tap on create shipping label
4. Select a package
6. Check that shipping rates are fetched from the API
7. Check that tapping on a shipping rate expands the shipping rate
8. Check that tapping on a signature required option selects the signature required option
9. 7. Check that tapping on a signature required option selected unselects the signature required option
10. Check that sorting keeps working as expected

### The tests that have been performed
1. Test that the UI works as expected 

### Images/gif

https://github.com/user-attachments/assets/5c384a7e-6ac6-465b-b06b-4dc38c373213



## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
